options/posix/generic/posix_signal.cpp

```diff
@@ -74,6 +74,20 @@ int sigaction(int signum, const struct sigaction *__restrict act, struct sigacti
 	return 0;
 }
 
+int siginterrupt(int sig, int flag) {
+    int ret;
+    struct sigaction act;
+
+    sigaction(sig, NULL, &act);
+    if (flag)
+        act.sa_flags &= ~SA_RESTART;
+    else
+        act.sa_flags |= SA_RESTART;
```

Comment:
Don't we need to clear the bit in any case?

Reply:
I dont understand, could you clarify?

Reply:
Oh, nevermind, I just had a mental lapse.